content/300-guides/200-deployment/201-serverless/300-deploy-to-vercel.mdx

```diff
@@ -6,281 +6,47 @@ metaDescription: 'Learn how to deploy a Next.js application based on Prisma Clie
+Vercel will automatically cache dependencies on deployment. For most applications, this will not cause any issues but for Prisma it may result in outdated clients. To avoid this issue, add `prisma generate` as a postinstall step to your application:
 
-### Result
-
-When the build completes, you should see the following:
-
-![Vercel deployment succeeded](./images/300-50-deploy-to-vercel-success.png)
-
-You have successfully deployed the app to Vercel. Note that Vercel will build and deploy further commits to your GitHub repository.
-
-To check that the API returns data successfully, click on the image of the deployed application. This will take you to your application at `https://__PROJECT-NAME__.vercel.app/`, where `__PROJECT-NAME__` is the name that Vercel generates for the project.
-
-When the page loads, click **Check API status**. If successful, you will see a result similar to the image below.
-
-```code no-copy
-{"up":true}
+```json file=package.json highlight=4;add
+{
+  ...
+  "scripts" {
+    "postinstall": "prisma generate"
+  }
+  ...
+}
 ```
 
-## 2. Test your deployed application
-
-You can now test your application. Open your application at `https://__PROJECT_NAME__.vercel.app` to interact with the API.
-
-### Steps
-
-1. Click **Load users with profiles**.
-
-   You do not yet have any data in the database, so the application returns an empty array:
-
-   ```code no-copy
-   []
-   ```
-
-   <Admonition type="info">
-
-   This API call uses Prisma to get all `users` in the database with their related `profiles`. The code is in the `pages/api/users.js` file in the project repository.
-
-   </Admonition>
-
-2. Click **Seed data**.
-
-   <Admonition type="info">
-
-   This API call uses Prisma to delete any existing database records and load the database with test `users`, `profiles`, and `posts`, then returns the created `users`. The code is in the `pages/api/seed.js` file.
+This will force prisma to re-generate the client on build so that your deployment always has an up to date client.
 
-   </Admonition>
+## Add a separate database for preview deployments
 
-3. Click **Load users with profiles** again.
+By default, your application will have a single _production_ environment, associated with the `main` Git branch of your repository. If you open a pull request to change your application, Vercel creates a new _preview_ environment.
 
-   This now returns the `users` and `profiles` that you added in the previous step:
+Vercel uses the `DATABASE_URL` environment variable that you defined when you imported the project for both the production and preview environments. This causes problems if you create a pull request with a database schema migration, because the pull request will change the schema of the production database.
```

Comment:
```suggestion
Vercel uses the `DATABASE_URL` environment variable you define when you import the project for both the production and preview environments. This causes problems if you create a pull request with a database schema migration because the pull request will change the schema of the production database.
```